app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -57,26 +53,20 @@ export async function showAccountCreationDialog(
         true,
       ),
     );
+    return confirmationResult;
   } catch (e) {
     throw new Error(
       `Error occurred while showing account creation dialog.\n${e}`,
     );
-  } finally {
-    if (addAccountApprovalId !== undefined) {
-      controllerMessenger.call('ApprovalController:endFlow', {
-        id: addAccountApprovalId,
-      });
```

Comment:
Does this flow get terminated by the confirmation system? If so, what was the reasoning we needed to manually end the approval flow [before](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/snap-keyring/snap-keyring.ts#L253-L257)?